### PR TITLE
Add FlexIO registers to Resources object

### PR DESCRIPTION
**Pull request by Finomnis**
All three FlexIO drivers are connected to pins that are available on the Teensy4.0.

Examples:
- Teensy 4.0 Pin `5` - `EMC_08` - Flexio1, Pin `8`
- Teensy 4.0 Pin `6` - `B0_10` - Flexio2, Pin `10`
- Teensy 4.0 Pin `14` - `AD_B1_02` - Flexio3, Pin `2`

This change is related to https://github.com/imxrt-rs/imxrt-iomuxc/pull/37.

A driver in `imxrt-hal` does not exist yet and most likely never will, because `FlexIO` is just simply too generic. So most likely there will be usecase-specific drivers in separate crates.

Like my usecase; I am in the process of writing a `ws2812-flexio` driver crate for Neopixel LED Strips.

### CODE REVIEW

**Comment by mciantyre** [comment]
Now that we have clock gates available in imxrt-hal, should we ensure that the FlexIO clock gates are on? This would be consistent with the board's clock policy.

I agree that this peripheral is flexible, so I'm indifferent about making `board` choose the root clocks and FlexIO input clock frequencies. If we want to skip this, we could document an exception for the FlexIO peripheral. Otherwise, we could choose default clocks and frequencies. It's OK to use imxrt-**ral** for clock changes until there's corresponding APIs in imxrt-**hal**.

https://github.com/mciantyre/teensy4-rs/blob/e6eb36d5e73cb018209a9836bd3da32a6c4dd4ef/src/board.rs#L54-L62

**Comment by Finomnis** [comment]
> Now that we have clock gates available in imxrt-hal, should we ensure that the FlexIO clock gates are on? This would be consistent with the board's clock policy.
> 

That would make sense if unconfigured, FlexIO does have default clock frequencies which are specified in the documentation. (see the black dot next to the muxes)

**Comment by Finomnis** [comment]
@mciantyre Done :)

File: Cargo.toml [hunk 1]
```diff
@@ -42,7 +42,7 @@ version = "0.5"
 features = ["imxrt1062"]
 
 [dependencies.imxrt-hal]
-version = "0.5"
+version = "0.5.3"
```
[the remark this answers is not in the document]
Comment (Finomnis) [reply Cargo.toml:45]: I'm quite certain that `0.5.3` already means exactly that. To prevent updates, you would need to add `=0.5.3`.

**Comment by Finomnis** [comment]
@mciantyre Any news on this? :)

**Review by mciantyre — APPROVED** [review]
Looks good. I added the clock policy exception and a changelog entry.

**Comment by Finomnis** [comment]
@mciantyre Great work! Thanks a lot for your collaboration. I highly appreciate the work you do for this community.

**Pull request merged**